parser/src/parser.rs

```diff
@@ -385,17 +386,53 @@ where
 
     fn parse_program(&mut self) -> ParserResult<'f, ast::Program<'f>> {
         spanned!(self, {
+            let mut attrs = Vec::new();
+            loop {
+                if let Ok(TokenKind::Operator(Operator::Slash)) = self.lexer.peek().map(|t| t.data)
+                {
+                    if let Ok(TokenKind::Operator(Operator::Slash)) =
+                        self.lexer.peek_nth(1).map(|t| t.data)
+                    {
+                        if let Ok(TokenKind::Operator(Operator::QuestionMark)) =
+                            self.lexer.peek_nth(2).map(|t| t.data)
+                        {
+                            if let Ok(TokenKind::Operator(Operator::Exclaim)) =
+                                self.lexer.peek_nth(3).map(|t| t.data)
+                            {
+                                self.eat(Exactly(TokenKind::Operator(Operator::Slash)))?;
+                                self.eat(Exactly(TokenKind::Operator(Operator::Slash)))?;
+                                self.eat(Exactly(TokenKind::Operator(Operator::QuestionMark)))?;
+                                self.eat(Exactly(TokenKind::Operator(Operator::Exclaim)))?;
+                                attrs.push(self.parse_attribute()?);
+                                continue;
```

Comment:
NIT: wouldn't this be more readable as a while loop?

Reply:
Nope, we need to peek 4 chars ahead before we can decide if it's a global attribute or if it belongs to a class